### PR TITLE
[Workflows] Fix run notifications

### PR DESCRIPTION
**Pull request by GiladShapira94**
Related tickets - 
https://iguazio.atlassian.net/browse/ML-6885
https://iguazio.atlassian.net/browse/ML-6884

This PR fix two issues - 
1. return run results in the notifications when running workflow
2. attached run results when using webhook notifications with override body

### CODE REVIEW

**Review by liranbg — CHANGES_REQUESTED** [review]
missing some unit tests

File: mlrun/utils/notifications/notification/webhook.py [hunk 1]
```diff
@@ -63,6 +63,33 @@ async def push(
             request_body["custom_html"] = custom_html
 
         if override_body:
+            list_edit_runs = []
+            for run in runs:
+                if isinstance(run, mlrun.model.RunObject):
```
Comment (alonmr) [inline mlrun/utils/notifications/notification/webhook.py]: Can also be RunTemplate
```suggestion
                if hasattr(run,"to_dict):
```

File: tests/utils/test_notifications.py [hunk 2]
```diff
@@ -196,6 +197,170 @@ def test_condition_evaluation_timeout():
...
+        "message": "runs: [{'project': 'test-remote-workflow', 'name': 'func-func', 'host': 'func-func-8lvl8', "
+        "'status': {'state': 'completed', 'results': {'return': 1}}}]"
+    }
+    requests_mock.assert_called_once_with(
+        "http://test.com", headers={}, json=expected_body, ssl=None
+    )
+
+
+# An example for a failed runs inputs for webhook notifications
+run_example = {
```
Comment (alonmr) [inline tests/utils/test_notifications.py]: do you need both this and the one above?
please create a function that generates a run example and if you need to change values accept them as parameters

File: mlrun/utils/notifications/notification/webhook.py [hunk 3]
```diff
@@ -63,6 +63,33 @@ async def push(
...
+                        "name": run["metadata"]["name"],
+                        "host": run["metadata"]["labels"]["host"],
+                        "status": {"state": run["status"]["state"]},
+                    }
+                    if run["status"].get("error", None):
+                        r["status"]["error"] = run["status"]["error"]
+                    elif run["status"].get("results", None):
+                        r["status"]["results"] = run["status"]["results"]
+                    run = r
+                list_edit_runs.append(run)
```
Comment (alonmr) [inline mlrun/utils/notifications/notification/webhook.py]: Not sure why you are appending a run if its not a dict but here is a suggestion to improve this block
```suggestion
                    parsed_run = {
                        "project": run["metadata"]["project"],
                        "name": run["metadata"]["name"],
                        "host": run["metadata"]["labels"]["host"],
                        "status": {"state": run["status"]["state"]},
                    }
                    if run["status"].get("error", None):
                        parsed_run["status"]["error"] = run["status"]["error"]
                    elif run["status"].get("results", None):
                        parsed_run["status"]["results"] = run["status"]["results"]
                    list_edit_runs.append(parsed_run)
```

File: mlrun/utils/notifications/notification/webhook.py [hunk 4]
```diff
@@ -63,6 +63,33 @@ async def push(
...
+            if isinstance(override_body, dict):
+                for key, value in override_body.items():
+                    if "{{ runs }}" in value:
+                        override_body[key] = value.replace(
+                            "{{ runs }}", str(list_edit_runs)
+                        )
+                    elif "{{runs}}" in value:
+                        override_body[key] = value.replace(
+                            "{{runs}}", str(list_edit_runs)
+                        )
```
Comment (alonmr) [inline mlrun/utils/notifications/notification/webhook.py]:
```suggestion
                runs_value = str(list_edit_runs)
                for key, value in override_body.items():
                    override_body[key] = value.replace(
                            "{{ runs }}", runs_value
                        ).replace(
                            "{{runs}}", runs_value
                        )
```

**Review by alonmr — COMMENTED** [review]
Round 1 🙂

File: tests/utils/test_notifications.py (hunk 2 above)
Comment (GiladShapira94) [reply tests/utils/test_notifications.py]: fixed, i created one function that used for those two tests

File: mlrun/utils/notifications/notification/webhook.py (hunk 4 above)
Comment (GiladShapira94) [reply mlrun/utils/notifications/notification/webhook.py]: thanks 💪  fixed

File: tests/utils/test_notifications.py [hunk 5]
```diff
@@ -196,6 +197,126 @@ def test_condition_evaluation_timeout():
     assert notification_pusher._should_notify(run, notification)
 
 
+def generate_run_result(succeed):
```
Comment (alonmr) [inline tests/utils/test_notifications.py]:
```suggestion
def _generate_run_result(state: str, error: str = None, results: dict = None):
```

**Review by alonmr — APPROVED** [review]

**Review by liranbg — APPROVED** [review]

**Pull request merged**